plenum/test/view_change/test_different_last_ordered_before_view_change.py

```diff
@@ -0,0 +1,154 @@
+import pytest
+import sys
+
+from plenum.server.node import Node
+from plenum.test.delayers import cDelay, pDelay, ppDelay
+from plenum.test.helper import sdk_send_random_and_check, \
+    sdk_send_random_requests, sdk_get_replies
+from plenum.test.stasher import delay_rules
+from stp_core.loop.eventually import eventually
+
+
+def test_different_last_ordered_on_backup_before_view_change(looper, txnPoolNodeSet,
+                                                  sdk_pool_handle, sdk_wallet_client):
+    ''' Send random request and do view change then fast_nodes (1, 4 - without
+    primary backup replicas) are already ordered transaction on master and some backup replica
+    and slow_nodes are not on backup replica. Wait ordering on slow_nodes.'''
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                              sdk_wallet_client, 1)
+    slow_instance = 1
+    slow_nodes = txnPoolNodeSet[1:3]
+    fast_nodes = [n for n in txnPoolNodeSet if n not in slow_nodes]
+    nodes_stashers = [n.nodeIbStasher for n in slow_nodes]
+    old_last_ordered = txnPoolNodeSet[0].replicas[slow_instance].last_ordered_3pc
+    with delay_rules(nodes_stashers, cDelay(delay=sys.maxsize,
+                                            instId=slow_instance)):
+        # send one  request
+        requests = sdk_send_random_requests(looper, sdk_pool_handle,
+                                            sdk_wallet_client, 1)
+        sdk_get_replies(looper, requests)
+        old_view_no = txnPoolNodeSet[0].viewNo
+        looper.run(
+            eventually(last_ordered,
+                       fast_nodes,
+                       (old_view_no, old_last_ordered[1] + 1),
+                       slow_instance))
+        last_ordered_for_slow = slow_nodes[0].replicas[slow_instance].last_ordered_3pc
```

Comment:
Please also check that  `last_ordered` on slow nodes (on slow instance) is  (old_view_no, old_last_ordered). 
Please check that `last_ordered` on all master instances is  (old_view_no, old_last_ordered+1).

Reply:
On backup instances last ordered is (new_view_no, 1) because with current logic we sending preprepare message without transactions. And order it. I think it's a very strange behavior. But for backup, may be it is normal.
Added check that last_ordered on all master instances is (old_view_no, old_last_ordered+1).

Reply:
Need to add a check of empty transaction list for sending preprepare?